pallets/poa/src/lib.rs

```diff
@@ -1127,8 +1135,16 @@ impl<T: Trait> Module<T> {
         print("slots_per_validator");
         print(slots_per_validator);
 
-        if slots_per_validator > max_blocks.to_number() {
-            panic!("This panic should never trigger");
+        // It might happen that `slots_per_validator` > `max_blocks` as the network went down for
+        // a brief moment of time but a validator should not be able to produce any more than 1 blocks
+        // more than the slots allowed per validator. The cushion of 1 is given in the case when network
+        // abruptly terminates and some validators don't get a chance to produce blocks
+        let max_bl = max_blocks.to_number();
+        if (max_bl > slots_per_validator) && ((max_bl - slots_per_validator) > 1) {
```

Comment:
The check was `slots_per_validator > max_blocks.to_number()` previously. Now it is `max_blocks.to_number() > slots_per_validator`. I see why the old check should be removed, but the new one doesn't quite make sense to me.

I think the new check is equivalent to:

```rust
max_blocks.to_number().saturating_sub(slots_per_validator) > 1
```

So are we now checking to make sure that a validator never produces more than one too many blocks?

Reply:
Yes, never produces more than one too many slots alloted